mantidproject/mantid
# Enable Mantid to support HFIR HB3A 512 x 512 detector

Description of work.

* Generated a new HB3A's IDF for its new 512 x 512 detector;
* Modified the beginning and expiration time of the original IDF of HB3A;
* Modified LoadSpiceXML2DDet such that it was able to load the instrument without user's specifying detector geometry.  A unit test was implemented for this feature.

**To test:**



1. Launch Mantidplot and then interface "HFIR 4Circle Reduction";
2. Test with Exp 566. You can choose arbitrary scans such as  1931-1941. 
3. Choose one or two scans to add peak.
4. Set up UB matrix by reading from SPICE file.
5. Index the peaks with UB matrix to check whether the results are consistent with SPICE.


Fixes #18896 .

*Does not need to be in the release notes.*
The new feature. will be described in release notes in the next PR associated with HB3A.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
